json-schema/json-schema-generic/src/main/scala/endpoints/generic/JsonSchemas.scala

```diff
@@ -120,29 +121,35 @@ trait JsonSchemas extends algebra.JsonSchemas {
   trait GenericJsonSchemaLowLowPriority {
 
     trait GenericRecord[A] extends GenericJsonSchema[A] {
-      def jsonSchema: Record[A]
+      def jsonSchema(docs: List[Option[documentation]]): Record[A]
     }
 
     trait GenericTagged[A] extends GenericJsonSchema[A] {
-      def jsonSchema: Tagged[A]
+      def jsonSchema(docs: List[Option[documentation]]): Tagged[A]
     }
 
-    implicit def recordGeneric[A, R](implicit
+    implicit def recordGeneric[A, R, D <: HList](implicit
       gen: LabelledGeneric.Aux[A, R],
       record: GenericRecord[R],
+      docAnns: Annotations.Aux[documentation, A, D],
+      docsToList: ToList[D, Option[documentation]],
       ct: ClassTag[A]
     ): GenericRecord[A] =
       new GenericRecord[A] {
-        def jsonSchema: Record[A] = nameSchema(record.jsonSchema.invmap[A](gen.from)(gen.to))
+        def jsonSchema(docs: List[Option[documentation]]): Record[A] =
+          nameSchema(record.jsonSchema(docsToList(docAnns.apply)).invmap[A](gen.from)(gen.to))
       }
 
-    implicit def taggedGeneric[A, R](implicit
+    implicit def taggedGeneric[A, R, D <: HList](implicit
       gen: LabelledGeneric.Aux[A, R],
       tagged: GenericTagged[R],
+      docAnns: Annotations.Aux[documentation, A, D],
+      docsToList: ToList[D, Option[documentation]],
       ct: ClassTag[A]
     ): GenericTagged[A] =
       new GenericTagged[A] {
-        def jsonSchema: Tagged[A] = nameSchema(tagged.jsonSchema.invmap[A](gen.from)(gen.to))
+        def jsonSchema(docs: List[Option[documentation]]): Tagged[A] =
```

Comment:
Unfortunately it involves somewhat weird runtime propagation like this. Maybe I should add a comment that we're ignoring the method argument on purpose or even an assertion to make sure that `docs == Nil`.